tests/test_security.py

```diff
@@ -0,0 +1,72 @@
+import pytest
+from fastapi import HTTPException
+
+from app.api.security import verify_token
+
+
+def test_missing_client_id_raises_error_when_auth_enabled(
+    monkeypatch, test_app
+):
+    """Test that a missing client ID raises an error on startup when authentication is enabled."""
+    # We're using what should be default values of CLIENT_ID and AUTH_ENABLED here
+    # (if the corresponding environment variables are unset),
+    # but we set the values explicitly here for clarity
+    monkeypatch.setattr("app.api.security.CLIENT_ID", None)
+    monkeypatch.setattr("app.api.security.AUTH_ENABLED", True)
+
+    with pytest.raises(ValueError) as exc_info:
+        with test_app:
+            pass
+
+    assert "NB_QUERY_CLIENT_ID is not set" in str(exc_info.value)
+
+
+# Ignore startup warning that is unrelated to the current test
+@pytest.mark.filterwarnings(
+    "ignore:No local Neurobagel nodes defined or found"
+)
+def test_missing_client_id_ignored_when_auth_disabled(monkeypatch, test_app):
+    """Test that a missing client ID does not raise an error when authentication is disabled."""
+    monkeypatch.setattr("app.api.security.CLIENT_ID", None)
+    monkeypatch.setattr("app.api.security.AUTH_ENABLED", False)
+
+    with test_app:
+        pass
+
+
+@pytest.mark.parametrize(
+    "invalid_token",
+    ["Bearer faketoken", "Bearer", "faketoken", "fakescheme faketoken"],
+)
+def test_invalid_token_raises_error(invalid_token):
+    """Test that an invalid token raises an error from the verification process."""
+    with pytest.raises(HTTPException) as exc_info:
+        verify_token(invalid_token)
+
+    assert exc_info.value.status_code == 401
+    assert "Invalid token" in exc_info.value.detail
+
+
+@pytest.mark.parametrize(
+    "invalid_auth_header",
+    [{}, {"Authorization": ""}, {"badheader": "badvalue"}],
+)
+def test_query_fails_with_malformed_auth_header(
+    test_app, monkeypatch, invalid_auth_header
+):
+    """Test that a request to the /query route fails with a missing or malformed authorization header."""
+
+    # Mock verify_token function since we don't want to actually verify the token in this test
+    def mock_verify_token(token):
+        return None
```

Comment:
couldn't you use the fixture with the same name here? It seems identical - but maybe you added the fixture later and forgot to remove here?

Reply:
good call!